physx/examples/scene_queries.rs

```diff
@@ -0,0 +1,258 @@
+            .unwrap(),
+        )
+        .unwrap();
+
+    let mut material = physics.create_material(0.5, 0.5, 0.6, ()).unwrap();
+
+    let ground_plane = physics
+        .create_plane(Vec3::new(0.0, 1.0, 0.0).into(), 0.0, material.as_mut(), ())
+        .unwrap();
+    // The scene owns actors that are added to it.  They can be retrieved using the
+    // various getters on the scene.
+    scene.add_static_actor(ground_plane);
+
+    let sphere_geo = PxSphereGeometry::new(10.0);
+    let sphere_pos = Vec3::new(0.0, 40.0, 100.0);
+
+    let mut sphere_actor = physics
+        .create_rigid_dynamic(
+            PxTransform::from_translation(&sphere_pos.into()),
+            &sphere_geo,
+            material.as_mut(),
+            10.0,
+            PxTransform::default(),
+            (),
+        )
+        .unwrap();
+    sphere_actor.set_angular_damping(0.5);
+    sphere_actor.set_rigid_body_flag(RigidBodyFlag::EnablePoseIntegrationPreview, true);
+    scene.add_dynamic_actor(sphere_actor);
+
+    // Do some scene queries
+    let origin = sphere_pos + Vec3::new(0.0, 100.0, 0.0);
+    let up_dir = Vec3::new(0.0, 1.0, 0.0);
+    let down_dir = -up_dir;
+    let distance = 5000.0;
+    unsafe {
+        // Raycasts
+        let mut hit = PxRaycastBuffer::new();
+        let hit_flags = HitFlag::default_hit_flags();
+        let filter_data = PxQueryFilterData::default();
+
+        let mut custom_hit = MyRaycastCallback::new();
+
+        let did_ray_hit = scene.raycast(
+            &origin.into(),
+            &down_dir.into(),
+            distance,
+            &mut hit,
+            hit_flags,
+            &filter_data,
+            None,
+            None,
+        );
+        assert!(did_ray_hit); // hits dynamic sphere in scene
+
+        // Let's see some info about the hit
+        let hit_info = hit.get_raycast_hit().unwrap();
+        dbg!(hit_info.distance);
+        // dbg!(PxVec3::from(hit_info.position));
+        // dbg!(PxVec3::from(hit_info.normal));
+
+        let did_ray_hit = scene.raycast(
+            &origin.into(),
+            &down_dir.into(),
+            distance,
+            &mut custom_hit,
+            hit_flags,
+            &filter_data,
+            None,
+            None,
+        );
+        assert!(did_ray_hit); // hits dynamic sphere in scene
+
+        let did_ray_hit = scene.raycast(
+            &origin.into(),
+            &up_dir.into(),
+            distance,
+            &mut hit,
+            hit_flags,
+            &filter_data,
+            None,
+            None,
+        );
+        assert!(!did_ray_hit); // raycast upward does not hit anything
+
+        // Overlaps
+        let mut overlap_callback = PxOverlapBuffer::new();
+        let did_overlap = scene.overlap(
+            &sphere_geo,
+            &PxTransform::default(),
+            &mut overlap_callback,
+            &filter_data,
+            None,
+        );
+        assert!(did_overlap); // overlaps with ground plane
```

Comment:
This assertion fails after getting this error from PhysX: `..\physx-rs\physx-sys\PhysX/physx\source/physx/src\NpSceneQueries.cpp (709) : invalid parameter : PxScene::overlap() and PxBatchQuery::overlap() calls without eANY_HIT flag require a touch hit buffer for return results.`
If I comment this assertion out, I get the same error again for the other overlap call, but sweep and raycast work great.